Unfolded/core/processing/layer2dataset.py

```diff
@@ -202,7 +202,7 @@ def _save_layer_to_file(layer: QgsVectorLayer, output_path: Path) -> Path:
         options = QgsVectorFileWriter.SaveVectorOptions()
         options.driverName = "csv"
         options.fileEncoding = "utf-8"
-        options.layerOptions = ["SEPARATOR=TAB", "STRING_QUOTING=IF_NEEDED"]
+        options.layerOptions = ["SEPARATOR=TAB", "STRING_QUOTING=ALWAYS"]
```

Comment:
Is this a built-in QGIS function?

Reply:
Presumably QGIS has its own CSV writer that this sets the configuration for? I haven't followed the discussion as closely, but we're fine still using a tab for the separator?

Reply:
It is [Ogr's CSV creation option](https://gdal.org/drivers/vector/csv.html#creation-issues) which is used by QGIS.

Reply:
And yes we are using tab as a separator since other possible (typical) separators (comma, semicolon, space) might be included in the data.

Reply:
That's why you use quoting...? If you couldn't use a comma in a CSV then it would be a horrible format. For example in Python (Pandas), the default behavior is to quote columns or values that need it:
```py
df = pd.DataFrame({'a': ['1', '2', ',']})
print(df.to_csv())
```
```
,a
0,1
1,2
2,","
```

Or, you can tell it to use string quoting always 
```py
import csv
df = pd.DataFrame({'a': ['1', '2', ',']})
print(df.to_csv(quoting=csv.QUOTE_ALL))
```
```
"","a"
"0","1"
"1","2"
"2",","
```
IMO that's still a much simpler solution than using tabs.

Reply:
Actually the quoting was because Unfolded Studio or kepler.gl cannot parse the files properly if the first column and first row containing data is numeric and the separator is something else than comma.

Reply:
We would of course use pandas (or geopandas in this case) and not be stucked with but in QGIS plugins you can only use standard library and the libraries QGIS and QT provide.